src/metabase/api/dashboard.clj

```diff
@@ -738,10 +738,14 @@
   [{source-options :source_options :as _param} query]
   (when-let [values (:values source-options)]
     {:values (if query
-               (parameter-card/query-matches query values)
+               (custom-values/query-matches query values)
```

Comment:
If `query` is nil then `query-matches` will return `values` anyway. Either remove the branching from here, or in the body of `query-matches`

Reply:
Oh, I just realised this problem was here before, you just moved query-matches. Even so, best tidy that up now

Reply:
I removed the branching in `query-matches`. having `query-matches` handle the branching seems to be a bit of magic if you're reading code from this namespace.

Reply:
You removed the branching already? I don't see it yet

Reply:
I removed the branch inside `query-matches` function, not here. Also moved the file to this ns.
https://github.com/metabase/metabase/pull/27343/commits/373301925d5902d51d0ab1f5e9f2b8f55e496f08